[Ajjagottu Kaushik Kumar Reddy] iP

kaushikkrdy opened the pull request:
# DukeBot

> “Not only should you have a to-do list, but it must become your best friend.” — Jim Kwik [(source)](https://medium.com/@oaklinejournal/25-quotes-from-successful-people-on-the-power-of-writing-to-do-lists-8a0b3729a34c)

DukeBot is a greenfield Java Project made in CS2103T (Software Engineering).
It's named after the Java mascot _Duke_. 

## About

DukeBot is a CLI (Command Line Interface) program that functions as a to-do list. 
It frees your mind of having to remember things you need to do. 
It's: 
* Easy to install 
* ~~Easy~~ **SUPER EASY** to use
* Text-based 

Start using it **today** to increase your productivity!

## Features 

Features:
* Managing tasks
* Managing deadlines
* Managing events
* Filter your tasks
* Save the date and time of your tasks


## Setting up in Intellij

Prerequisites: JDK 11, update Intellij to the most recent version.

1. Open Intellij (if you are not in the welcome screen, click `File` > `Close Project` to close the existing project first)
1. Open the project into Intellij as follows:
   1. Click `Open`.
   1. Select the project directory, and click `OK`.
   1. If there are any further prompts, accept the defaults.
1. Configure the project to use **JDK 11** (not other versions) as explained in [here](https://www.jetbrains.com/help/idea/sdk.html#set-up-jdk).<br>
   In the same dialog, set the **Project language level** field to the `SDK default` option.
3. After that, locate the `src/main/java/Duke.java` file, right-click it, and choose `Run Duke.main()` (if the code editor is showing compile errors, try restarting the IDE). If the setup is correct, you should see something like the below as the output:
   ```
   Hello! I'm Duke.
   What can I do for you?
   ```

## For programmers
If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```java 
public Duke {
    public static void main(String[] args) {
        new Duke("data/duke.txt").run();
    }
}
```

`data/duke.txt` is where your tasks will be saved and retrieved from when needed.

Commit:
Add Gradle support

kaushikkrdy committed:
Level-1

kaushikkrdy committed:
Level-2

kaushikkrdy committed:
Level-3

kaushikkrdy committed:
Level-4

kaushikkrdy committed:
A-TextUiTesting

kaushikkrdy committed:
Level-5

kaushikkrdy committed:
Level-6

kaushikkrdy committed:
A-Enums

kaushikkrdy committed:
Refactor Duke main method to include run()

kaushikkrdy committed:
Modify Task class into abstract class

kaushikkrdy committed:
Add loading and saving of tasks

kaushikkrdy committed:
Add ability to read strings as LocalDateTime objects

kaushikkrdy committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Add loading and saving of tasks

kaushikkrdy committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Add ability to read strings as LocalDateTime objects

# Conflicts:
#	src/main/java/Duke.java

kaushikkrdy committed:
Update runtest.sh to test for delete, done, new DateTime strings and new error messages

kaushikkrdy committed:
Extract Duke class code into Ui, Storage, Parser and TaskList classes

kaushikkrdy committed:
Fix issue where 'bye' command does not exit program

kaushikkrdy committed:
Package classes into duke package and update runtest.sh accordingly

kaushikkrdy committed:
Add unit test for Event class initialisation and Parser class command parsing exceptions

kaushikkrdy committed:
Add metadata about iP jar

kaushikkrdy committed:
Add header comments to non-private classes/methods and non-trivial private methods.

kaushikkrdy committed:
Add ability to find task by searching for a keyword

kaushikkrdy committed:
Tweak code to comply with Java coding standard

kaushikkrdy committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Add header comments to non-private classes/methods and non-trivial private methods.

kaushikkrdy committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Tweak code to comply with Java coding standard

# Conflicts:
#	src/main/java/duke/Deadline.java
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/Todo.java
#	src/main/java/duke/Ui.java

kaushikkrdy committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Add ability to find task by searching for a keyword

# Conflicts:
#	src/main/java/duke/CommandType.java
#	src/main/java/duke/Duke.java
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/Ui.java

kaushikkrdy committed:
Update runtest.sh to test for find functionality

kaushikkrdy committed:
Tweak header comments to fit Java coding standards

leezhixuan commented on src/main/java/duke/DukeException.java:
```diff
@@ -0,0 +1,11 @@
+    public DukeException(String message) {
+        super(message);
+    }
+}
```
Do you think that the description of the class fits the constructor method better?

leezhixuan commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,202 @@
+                    "D/MM/YYYY HHMM");
+        } catch (StringIndexOutOfBoundsException e) {
+            throw new DukeException("Please write your event command in the following format: " +
+                    "event task /at datetime");
```
Shouldn't the trailing "+" be moved to the start of the next line?

leezhixuan commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,40 @@
+        return (isDone? "[X]" : "[ ]");
+    }
+
+    public abstract String getFileString();
```
Do you think javadocs should be added for this method?

leezhixuan reviewed (commented):
I think your work is really neat and well documented. Good work!!

kaushikkrdy committed:
Update README

kaushikkrdy committed:
Tweak code to fit Java coding standards

kaushikkrdy commented:
> I think your work is really neat and well documented. Good work!!

Have taken your suggestions into consideration and have made the appropriate changes to my code. Thank you for your feedback!

kaushikkrdy committed:
Merge commit 'd8398594b7bc43da5eb865321c5a50cec4b3923d'

* commit 'd8398594b7bc43da5eb865321c5a50cec4b3923d':
  Add Gradle support

kaushikkrdy committed:
Modify build.gradle file to reflect correct mainClassName

szelongq commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,45 @@
+        this.isDone = false;
+    }
+
+    public void setIsDone(boolean isDone) {
```
Perhaps naming the method "setDone" would sound more intuitive to the user?

szelongq commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,45 @@
+    }
+
+    public String getStatusIcon() {
+        return (isDone? "[X]" : "[ ]");
```
Maybe adding a space between isDone and ? would make it easier to read?

szelongq reviewed (commented):
LGTM :)
Just two nits to fix. 
Great work!

kaushikkrdy commented:
> LGTM :)
> Just two nits to fix.
> Great work!

Have taken your suggestions into consideration and have made the appropriate changes to my code. Thank you for your feedback!

kaushikkrdy committed:
Modify boolean setter and added whitespace to fit with Java coding standards

kaushikkrdy committed:
Modify setter name to fit with Java coding standards

kaushikkrdy committed:
Add GUI

kaushikkrdy committed:
Merge branch 'branch-Level-10'

* branch-Level-10:
  Add GUI

kaushikkrdy committed:
Tweak header comments to fit modified methods in line with Java coding standards

kaushikkrdy committed:
Modify code to fit with Java checkstyle rules

kaushikkrdy committed:
Customise GUI by modifying appearance of dialog boxes and main window

kaushikkrdy committed:
Change mainClassName to Launcher class

kaushikkrdy committed:
Add assertions

kaushikkrdy committed:
Change assert statement to use equals() instead of ==

Comparing strings should be done by equals() instead of ==.

Comparing strings for their value equality should be done by using equals(), as == is used to check if they have the same reference.

Let's update the assert statement and replace all instances of == with equals() method.

kaushikkrdy committed:
Replace for loop with Java streams

kaushikkrdy committed:
Merge branch 'branch-A-Streams'

* branch-A-Streams:
  Replace for loop with Java streams

kaushikkrdy committed:
Tweak code to fit with coding quality standards

Code is not of high coding quality.

Let's,
* delete dead code that is not being used anymore
* prevent long methods (> 30 LoC) by refactoring code
* remove magic numbers by replacing them with named constants
* refactor code to keep to a single level of abstraction

kaushikkrdy committed:
Merge pull request #1 from kaushikkrdy/branch-A-Assertions

Add assertions

kaushikkrdy committed:
Merge branch 'master' into branch-A-CodeQuality

* master:
  Replace for loop with Java streams
  Change assert statement to use equals() instead of ==
  Add assertions

# Conflicts:
#	src/main/java/duke/MainWindow.java

kaushikkrdy committed:
Merge pull request #5 from kaushikkrdy/branch-A-CodeQuality

Code is not of high coding quality.

Let's,
* delete dead code that is not being used anymore
* prevent long methods (> 30 LoC) by refactoring code
* remove magic numbers by replacing them with named constants
* refactor code to keep to a single level of abstraction

kaushikkrdy committed:
List numbering: make numbering start from 1

List numbering should start from 1.

Starting from 1 is more intuitive and user-friendly than starting from 0.

Let's,
* update the IntStream code in the list() and find() methods to start from 1

kaushikkrdy committed:
Help command: add help command

Help command is missing.

A help command to show help message would be user-friendly as new users
will not know any of the commands.

Let's,
* add new CommandType HELP
* show help message when user types in "help"
* show instructions on how to use each command in help message
* set minimum height of dialog box to "-Infinity" for large help message

kaushikkrdy committed:
Replace default pictures with robot and user picture

kaushikkrdy committed:
Replace robot picture with one of smaller size

kaushikkrdy committed:
Add credits for reused work

kaushikkrdy committed:
Change user picture and provide picture attribution

kaushikkrdy committed:
Modify acknowledgements section of README

kaushikkrdy committed:
Add Ui.png to docs

kaushikkrdy committed:
Update user guide in docs/README.md

ramapriyan912001 commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,101 @@
+    @FXML
+    private ImageView displayPicture;
+
+    private final int TOP_PADDING = 5;
```
Great job making these constants instead of using magic literals.

ramapriyan912001 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,83 @@
+    private TaskList tasks;
+    private Ui ui;
+
+    private final int FIND_INPUT_START_INDEX = 5;
```
Great job using constants here instead of magic literals!

ramapriyan912001 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,83 @@
+     * @param userInput
+     * @return appropriate message based on user's input
+     */
+    public String run(String userInput) {
```
Very intuitive and easy to understand run method, great job!

ramapriyan912001 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,87 @@
+     * @throws DukeException if there is an invalid command or an incomplete command.
+     */
+
+    public static CommandType parse(String command) throws DukeException {
```
Good to see you have handled different case of commands too!

ramapriyan912001 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,222 @@
+     * @throws DukeException if description contains incorrectly formatted date/time or does not follow the format:
+     * event task /at datetime
+     */
+    public String createEvent(String description, Storage storage, Ui ui) throws DukeException {
```
Method could be slap-ed harder perhaps?

ramapriyan912001 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,222 @@
+     * @throws DukeException if description contains incorrectly formatted date/time or does not follow the format:
+     * deadline task /by datetime
+     */
+    public String createDeadline(String description, Storage storage, Ui ui) throws DukeException {
```
Method could be slap-ed harder perhaps?

ramapriyan912001 reviewed (commented):
Great job overall Kaushik! Not much apart from minor things to improve!

kaushikkrdy committed:
Update user guide

kaushikkrdy committed:
Update user guide with task examples

kaushikkrdy committed:
Tweak code to fit coding standards

kaushikkrdy committed:
Tweak code to fit coding quality
